pkg/engine/health/health.go

```diff
@@ -83,6 +70,10 @@ func IsHealthy(obj runtime.Object) error {
 			log.Printf("HealthUtil: Job \"%v\" is marked healthy", obj.Name)
 			return nil
 		}
+		if terminal, msg := isJobTerminallyFailed(obj); terminal {
+			return fmt.Errorf("%wHealthUtil: Job \"%v\" has failed terminally: %s", engine.ErrFatalExecution, obj.Name, msg)
```

Comment:
Here, we use `engine.ErrFatalExecution` marker to denote health errors that are likely not recoverable. The caller may or may not act upon this information.

Reply:
The only thing I don't like about it is that `ErrFatalExecution` is defined as  `errors.New("fatal error: ")` and that `fatal error: ` string breaks error message readability.